packages/components/src/components/time-picker/_time-picker.scss

```
@@ -64,6 +64,21 @@
       color: $text-05;
     }
   }
+
+  .#{$prefix}--time-picker--light .#{$prefix}--time-picker__select {
```

The styles set here are being overridden by the actual input styles themselves. 

```suggestion
  .#{$prefix}--time-picker--light .#{$prefix}--select-input {
```